File: CalculoIVAApp C++/Articulo.cpp
#include "Articulo.h"

Articulo::Articulo(const std::string& nombre, double precio, double porcentajeIVA)
    : nombre(nombre), precio(precio), porcentajeIVA(porcentajeIVA) {}

Articulo::~Articulo() {}

std::string Articulo::getNombre() const {
    return nombre;
}

void Articulo::setNombre(const std::string& nombre) {
    this->nombre = nombre;
}

double Articulo::getPrecioBase() const {
    return precio;
}

double Articulo::getParteIVA() const {
    return (precio * porcentajeIVA) / 100.0; // Calcula el IVA
}

double Articulo::getPrecioTotal() const {
    return precio + getParteIVA(); // Precio base + IVA
}

File: CalculoIVAApp/src/Articulo.java
public class Articulo {
    private String nombre; // Nombre del producto
    private double precio; // Precio sin IVA

    // Constructor
    public Articulo(String nombre, double precio) {
        this.nombre = nombre;
        this.precio = precio;
    }

    // Métodos de acceso y modificación para el nombre
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    // Método de acceso para el precio
    public double getPrecio() {
        return precio;
    }

    public void setPrecio(double precio) {
        this.precio = precio;
    }
}
